### ISSUE TITLE
Flatpak issues - crashing JASP 

### ISSUE THREAD

**Issue by JorisGoosen** [jasp-stats/jasp-issues#1424]
tl;dr
Workaround is as follows:
```
flatpak --user update --commit=4820a00d9526d39b6a370476bdba08843557d24efa9eff0dc70b401a57b5bd09 org.kde.Platform/x86_64/5.15
```

This will install a version of the kde platform from `build of org.kde.Sdk, Thu 26 Aug 2021 12:59:37 PM UTC (cb705061cdcb8ba73d87d310166bca55fe366e1e)`.

If you build locally you might want to downgrade the sdk also, as:
```
flatpak --user update --commit=82032c2cfb7f0f2cf55c66a98340bbc84aa61bc417e0874e8446d8c265fa40b5 org.kde.Platform/x86_64/5.15
```

### Why is the workaround needed?

It turns out that some newer updates to `org.kde.Platform` (which we use because it contains Qt, amongst other things) and the partner sdk `org.kde.Sdk` caused problems for JASP.

Such as:
#1165 #1378 https://github.com/jasp-stats/jasp-test-release/issues/1443 and more

Sadly enough I cannot debug it properly right now because of https://bugs.kde.org/show_bug.cgi?id=442422 where there are no debugsymbols available.

**Comment by jbingram** [comment]
Error: While pulling app/org.jaspstats.JASP/x86_64/stable from remote flathub: Server returned status 404: Not Found

**Comment by JorisGoosen** [comment]
Can you reach `flathub.org` in the webbrowser?

**Comment by JoKeyser** [comment]
I hit the 404 like @jbingram just when trying to `flatpak update` as well (and yes, flathub.org is reachable in principle). It seems to me that flathub is a bit flaky, it 404's at different stages of the download (sometimes it stops at 9%, 12%, or at 23%).
Here one example:
```
$ flatpak update -v org.jaspstats.JASP
[... stuff...]
1.	   	org.jaspstats.JASP	stable	u	flathub	< 1,3 GB
Proceed with these changes to the system installation? [Y/n]: 
Updating…                                                                                         F: Calling system helper: GetRevokefsFd
F: Calling system helper: GetRevokefsFd
F: flatpak_dir_pull: Using commit 67d5115263f38a2b490685f4458afaed2eeaf85b7de3486ab36a07523602e090 for pull of ref app/org.jaspstats.JASP/x86_64/stable from remote flathub
Updating… ██▍                   12%  662,9 kB/s  02:26F: Calling system helper: CancelPull
F: Calling system helper: CancelPull
Error: While pulling app/org.jaspstats.JASP/x86_64/stable from remote flathub: Server returned status 404: Not Found
Updates complete.                                                                                   
error: There were one or more errors
```

**Comment by JorisGoosen** [comment]
Well, I have no clue why you are running into 404 errors?

I just tried installing JASP from flathub, from scratch (as in, not updating a previous one) and it can find everything it needs and installs fine.

Have you tried first uninstalling all versions of JASP you have and then trying it again?

**Comment by jbingram** [comment]
$ flatpak remove JASP
Uninstall complete.
$ flatpak install JASP
Looking for matches…
Found ref ‘app/org.jaspstats.JASP/x86_64/stable’ in remote ‘flathub’ (system).
Use this ref? [Y/n]: 

org.jaspstats.JASP permissions:
    ipc       network              wayland              x11
    dri       file access [1]      dbus access [2]

    [1] home, xdg-config/kdeglobals:ro
    [2] com.canonical.AppMenu.Registrar

                ID                       Branch  Op   Remote   Download
 1. [✗] org.jaspstats.JASP     stable    i     flathub    1,0 kB / 1,3 GB

Error: Error pulling from repo: While pulling app/org.jaspstats.JASP/x86_64/stable from remote flathub: fsck 85e1c55464454e4d83846e12c841567d431fa22c806aa3d70d330bbd8cfa89ec.dirtree: No such metadata object 85e1c55464454e4d83846e12c841567d431fa22c806aa3d70d330bbd8cfa89ec.dirtree

error: Failed to install org.jaspstats.JASP: Error pulling from repo: While pulling app/org.jaspstats.JASP/x86_64/stable from remote flathub: fsck 85e1c55464454e4d83846e12c841567d431fa22c806aa3d70d330bbd8cfa89ec.dirtree: No such metadata object 85e1c55464454e4d83846e12c841567d431fa22c806aa3d70d330bbd8cfa89ec.dirtree

**Comment by JorisGoosen** [comment]
Can you try out `flatpak repair` as described: https://docs.flatpak.org/en/latest/flatpak-command-reference.html#flatpak-repair

Perhaps something went wrong locally?
Although your error looks suspicously similar to the problem I had with `org.kde.Sdk.Debug` as alluded to in https://bugs.kde.org/show_bug.cgi?id=442422 (which is why I can't solve the crash problem jasp has, as I cant debug it)

**Comment by JoKeyser** [comment]
The `flatpak repair` didn't work for me, but after uninstalling and then re-installing, it finally worked! So it may have been some local problem.

**Comment by tomtomme** [comment]
Confirmed. I had beta and stable on manjaro. Stable would not install. After removing both, stable installs fine.

**Comment by jbingram** [comment]
Weird. The procedure "remove JASP stable -> repair ->  install" was not working this morning and yesterday on 3 computers.
Just retried now and everything works on all computers.
NB: I had no beta version installed
Thanks for your help guys. Much appreciated

**Comment by JorisGoosen** [comment]
I guess flathub itself had a problem and they've now solved it? Or it took a while for them to process the new version...
Good to hear it works now though

**Comment by marodon** [comment]
Hello can not run JASP 0.16 on Ubuntu 20.04. I Install full lib canberra pakage
 won't open here are the messages (I tried uninstall/reinstall, same message. Is there a conficllt with NVDIA driver? I have a NVIDIA Corporation GP107 [GeForce GTX 1050 Ti] and I can't change the driver for reasons I don't know
gilles@gilles-MS-7970:~$ flatpak run org.jaspstats.JASP
Making new argument list for Application startup: org.jaspstats.JASP
Starting JASP 0.16 from commit 76a7260bf40277d6bf943c161ad7ba579f36b4b8 and branch HEAD
Gtk-Message: 18:28:24.146: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:28:24.146: Failed to load module "canberra-gtk-module"
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile)
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile)
Could not initialize GLX

**Comment by JorisGoosen** [comment]
Hey @marodon, Ive taken the liberty to open a new issue for you as what you describe is quite a different issue then the one at play here. I will respond to you there.

**Comment by JorisGoosen** [comment]
There has been another problem with flatpak on chromebook which isn't related to this one but it can't hurt to link to it: https://github.com/jasp-stats/jasp-issues/issues/1600

**Comment by Poker-Kid** [comment]
Dear people,
what can I do, if it gives me this:

error: org.kde.Platform/x86_64/5.15 not installed

while trying any of these comments?
Thank you very much in advance !
Jonathan

**Comment by tomtomme** [comment]
Try them with a sudo at the start

**Comment by Poker-Kid** [comment]
> Try them with a sudo at the start

Thank you, I were able to fix it - I think it worked after I reinstalled it with a special version.

**Comment by nothik** [comment]
Thanks a lot Joris!

I got the error message 
> error: org.kde.Platform/x86_64/5.15 not installed

but removing "--user" from the tl;dr command did the trick.

**Comment by tomtomme** [comment]
> Hi guys, sorry for the delay. My problem is exactly the one described by @JoKeyser . Sometimes stops at 1%, 5%, ..., 12%??? (wow never been that far). As of today, I still cannot update it. NB1:I can reach the flathub page on my web browser. NB2: this is the only flatpak software I have problem with, among the > 50 others installed on my computer. Cheers guys

For flatpak beta 0.16.3 the 1%, 5%, ..., 12% Problem appeared again. 
I fixed it by first uninstalling the old jasp-beta and then a fresh install of jasp beta. Maybe just add a note to the flatpak-beta installation guide to first uninstall old versions to close this issue.

**Comment by tomtomme** [comment]
Since today I get a new 0.16.3 debug-build through the **stable** flatpak channel - is this intended?
To successfuly install it, I had again to remove the old 0.16.1-stable first, only then 0.16.2-debug-stable would download and install.
![grafik](https://user-images.githubusercontent.com/4428185/174029188-cd37bc2c-9302-4ab6-83e1-3ef6cb018ed1.png)

**Comment by JorisGoosen** [comment]
Too bad the upgrade didnt work smoothly but at least the new build works right?

So this issue can finally be closed

**Comment by tomtomme** [comment]
I guess so. The build works. But why do we have a debug-build on flatpaks stable channel?

**Comment by JorisGoosen** [comment]
Well, the flatpak is always a debugbuild, but the debugsymbols are split off.

I guess that the name put in there is dependent on that debug-flag being on and not on those symbols being loaded.

Im checking it now, but I don't think the symbols are included, so it is just a bit of misleading text

**Comment by JorisGoosen** [comment]
Yeah so the debugsymbols are stored in `flatpak install flathub org.jaspstats.JASP.Debug`

But ill fix jasp so it won't say that it is a "debug-version" anymore

**Comment by utkb** [comment]
I have same 404 problem and this page helped me: https://github.com/flathub/flathub/issues/2927 

I used this code to update:

`flatpak update --no-static-deltas`

**Comment by nothik** [comment]
Thanks utkb!
Note to stick to the bottom of your screen: before trying to update JASP, visit issue 1424 to see what the current workaround is :)
